### ISSUE TITLE
[INF] Serve  NonTLS On: [port] with version 1.2.8+

### ISSUE THREAD

**Issue by nicoroy2561** [lesismal/nbio#146]
Hi,
First of all, thank you as always for the amazing work.

I have been holding back from updating nbio to the latest version due to an issue I'm encountering:
whenever I run nbio on version > the current (currently on 1.2.6, tested on 1.2.8 and 1.2.9) I get the following:
"[INF] Serve  NonTLS On: [:port]"
despite it working withouth any issue on version 1.2.6. It seems newer versions either have issues reading the TLS certificates, or something else is going on.

**Comment by lesismal** [comment]
Please provide a full example code that can reproduce the problem, and I'll look into it.

**Comment by nicoroy2561** [comment]
The code that instanciates the ws server is the following:

```
import(
....
	nbhtls "github.com/lesismal/llib/std/crypto/tls"
	"github.com/lesismal/nbio/nbhttp"
	"github.com/lesismal/nbio/nbhttp/websocket"
)

func runServer(handler HttpHandler) {
mux := &http.ServeMux{}
mux.HandleFunc(v.Pattern, handler)

cert, err := nbhtls.LoadX509KeyPair("fullchain.pem", "privkey.pem")
if err != nil {
	log.Fatalf("tls.LoadX509KeyPair failed: %v", err)
}
tlsConfig := &nbhtls.Config{
	Certificates:       []nbhtls.Certificate{cert},
	InsecureSkipVerify: false,
	MinVersion: nbhtls.VersionTLS12,
	PreferServerCipherSuites: true,
	CurvePreferences: []nbhtls.CurveID{
		nbhtls.X25519,
		nbhtls.CurveP256,
},

nbioServer := nbhttp.NewServerTLS(
	nbhttp.Config{
	Network:                 "tcp",
	Addrs:                   []string{portNumber},
	MaxLoad:                 100000,
	ReleaseWebsocketPayload: true,
},
http.HandlerFunc([callback]), nil, tlsConfig)

err := nbioServer.Start()
if err != nil {
	log.Print(1, "nbio.Start failed:", err) //internal logging package call
	return
}
}

```

Now, what I see up until v1.2.6 is the following output:
```
[INF] Gopher[NB] start listen on: [":[portNumber]"]
```
What I see when I update nbio to v1.2.8/1.2.9 (and llib to 1.1.4 from 1.1.2) is the following:
```
[INF] Gopher[NB] start
[INF] Serve  NonTLS On: [:[portNumber]]
```

And the clients aren't able to reach it.

**Comment by lesismal** [comment]
Your log shows that your code is `Serve NonTLS`, but usually it should `Serve TLS On:`, but I can't run your code.
I tried my examples and it runs fine, so I'm not sure what's really happening in your code, would you please provide the full code that can reproduce your problem? Or refer to the examples:
https://github.com/lesismal/nbio-examples

**Comment by nicoroy2561** [comment]
Thank you for your answer. 
I put together the following and tested:
```
package main

import (
	"fmt"
	"net/http"

	nbhtls "github.com/lesismal/llib/std/crypto/tls"
	"github.com/lesismal/nbio/nbhttp"
)

func main() {
	runServer()
}

func runServer() {
	cert, err := nbhtls.LoadX509KeyPair("fullchain.pem", "privkey.pem")
	if err != nil {
		fmt.Println("tls Load key pair error:", err)
		return
	}
	tlsConfig := &nbhtls.Config{
		Certificates:             []nbhtls.Certificate{cert},
		InsecureSkipVerify:       false,
		MinVersion:               nbhtls.VersionTLS12,
		PreferServerCipherSuites: true,
		CurvePreferences: []nbhtls.CurveID{
			nbhtls.X25519,
			nbhtls.CurveP256,
		},
	}
	nbioServer := nbhttp.NewServerTLS(
		nbhttp.Config{
			Network:                 "tcp",
			Addrs:                   []string{"1001"},
			MaxLoad:                 100000,
			ReleaseWebsocketPayload: true,
		},
		http.HandlerFunc(exampleHandler), nil, tlsConfig)

	err = nbioServer.Start()
	if err != nil {
		fmt.Println("nbio Start failed:", err.Error())
		return
	}
}

func exampleHandler(w http.ResponseWriter, req *http.Request) {
	fmt.Println("handling request")
}
```

This seemed to work - I wasn't getting the NonTLS line.
Though I'm not sure that is the intended behaviour - is it normal that I only get `[INF] Gopher[NB] start
` and never the `Serve TLS On:` you mentioned?
Do let me know if you can get the code above to run.

**Comment by lesismal** [comment]
> Thank you for your answer. I put together the following and tested:
> 
> ```
> package main
> 
> import (
> 	"fmt"
> 	"net/http"
> 
> 	nbhtls "github.com/lesismal/llib/std/crypto/tls"
> 	"github.com/lesismal/nbio/nbhttp"
> )
> 
> func main() {
> 	runServer()
> }
> 
> func runServer() {
> 	cert, err := nbhtls.LoadX509KeyPair("fullchain.pem", "privkey.pem")
> 	if err != nil {
> 		fmt.Println("tls Load key pair error:", err)
> 		return
> 	}
> 	tlsConfig := &nbhtls.Config{
> 		Certificates:             []nbhtls.Certificate{cert},
> 		InsecureSkipVerify:       false,
> 		MinVersion:               nbhtls.VersionTLS12,
> 		PreferServerCipherSuites: true,
> 		CurvePreferences: []nbhtls.CurveID{
> 			nbhtls.X25519,
> 			nbhtls.CurveP256,
> 		},
> 	}
> 	nbioServer := nbhttp.NewServerTLS(
> 		nbhttp.Config{
> 			Network:                 "tcp",
> 			Addrs:                   []string{"1001"},
> 			MaxLoad:                 100000,
> 			ReleaseWebsocketPayload: true,
> 		},
> 		http.HandlerFunc(exampleHandler), nil, tlsConfig)
> 
> 	err = nbioServer.Start()
> 	if err != nil {
> 		fmt.Println("nbio Start failed:", err.Error())
> 		return
> 	}
> }
> 
> func exampleHandler(w http.ResponseWriter, req *http.Request) {
> 	fmt.Println("handling request")
> }
> ```
> 
> This seemed to work - I wasn't getting the NonTLS line. Though I'm not sure that is the intended behaviour - is it normal that I only get `[INF] Gopher[NB] start ` and never the `Serve TLS On:` you mentioned? Do let me know if you can get the code above to run.

There seems no difference using nbio compared with the previous code you provide, so I still do not know what happened in your old code and your env.
`Serve TLS On` abd `Serve NonTLS On` is just log to show your service type, it is normal and depends on your server type.
I do not have your certi files, so I run my examples and I can get the right result succesful:
https://github.com/lesismal/nbio-examples/tree/master/http/server_tls

**Comment by nicoroy2561** [comment]
The old code was a bad example taken from my codebase, please ignore that.
This second one is a properly made example code to test.

When you run the code I gave you with any key pairs, do you get the line
Serving TLS on: ?
Or the second code I gave you doesn't run with your key pairs?
Thank you for your time as usual.

**Comment by lesismal** [comment]
Well I got it, here is a compatibility bug for the `config.Addrs` field in the new version.
In the new version, we implemented http/websocket client, rename the previous `Server` struct to `Engine` struct and left a `Server` struct which is simply wrap `Engine` to handle compatibility. We support both tls and non-tls addrs, and support both server-side and client-side connections in the same Engine.
Now when we use tls, config it like this:
```golang
nbhttp.Config{
    AddrsTLS: []string{...}, // not Addrs
}
```
That's the problem, it's not compatible with the previous version.
I'll make a new commit to fix it.

**Comment by lesismal** [comment]
Please try the [latest code](https://github.com/lesismal/nbio/commit/a6e6816c5d927fd1b324f6b3ad6ee788131e4333):
```sh
go get -u github.com/lesismal/nbio@a6e6816
```

Now it should runs fine like this:
```sh
go run ./server.go 

# Output
2021/12/05 09:45:25.791 [INF] Gopher[NB] start
2021/12/05 09:45:25.791 [INF] Serve     TLS On: [:1001]
```

**Comment by nicoroy2561** [comment]
Yup, I can confirm that was indeed the issue. Makes sense it wouldn't work any longer then.
Thanks a lot for the patience and the help :) Keep up the good work!

**Comment by lesismal** [comment]
Thank you for your feedback!

I have published v1.2.10:
https://github.com/lesismal/nbio/releases/tag/v1.2.10
